arch/riscv/src/lib.rs

```diff
@@ -45,105 +45,102 @@ extern "C" {
     static __global_pointer: usize;
 }
 
+/// Entry point of all programs (`_start`).
+///
+/// This assembly does three functions:
+///
+/// 1. It initializes the stack pointer, the frame pointer (needed for closures
+///    to work in start_rust) and the global pointer.
+/// 2. It initializes the .bss and .data RAM segments. This must be done before
+///    any Rust code runs. See https://github.com/tock/tock/issues/2222 for more
+///    information.
+/// 3. Finally it calls `main()`, the main entry point for Tock boards.
 #[cfg(any(doc, all(target_arch = "riscv32", target_os = "none")))]
-extern "C" {
-    // Entry point of all programs (`_start`).
-    ///
-    /// This assembly does three functions:
-    ///
-    /// 1. It initializes the stack pointer, the frame pointer (needed for closures
-    ///    to work in start_rust) and the global pointer.
-    /// 2. It initializes the .bss and .data RAM segments. This must be done before
-    ///    any Rust code runs. See <https://github.com/tock/tock/issues/2222> for more
-    ///    information.
-    /// 3. Finally it calls `main()`, the main entry point for Tock boards.
-    pub fn _start();
+#[link_section = ".riscv.start"]
+#[export_name = "_start"]
```

Comment:
Do we need this? I tried building hifive1 without it and it worked.
```suggestion
```

Reply:
I assumed we had this before for some valid reason so I kept it in, but maybe it was just due to a now-fixed bug with naked_functions? Or maybe it was for ease of debugging in gdb of some sort?

Reply:
happy to remove these, just want to keep this open to see if anyone can chime in on the original reasoning for having it